### PR TITLE
Refreshed and improved to support config less repos, key libs and be more resilient

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
New huge set of enhancements to the project, following #143 with main goal to make this usable on project with no prior config, and handle command lib with adherence to node version (lerna and serverless).
Also program was improved not to require a clean state before running.
Less config is needed, the program can be run out of the box on any node repository

------
Exemple of command ran: (that boostrapped that PR: https://github.com/CoorpAcademy/coorpacademy-lambda/pull/1590)
```bash
update-node upgrade --target 20 -dcam "Update-node major bump :outbox_tray:" -R @Coorpacademy/development-mooc -F api-media -s media
```
flags:
- `d` for default config :new: 7eb2685690183af3d3dead0270f2824d6083b634
- `c` for clean, cleanup before running :new: c6e0220e9614976d7a68a68db29a960d30464907
- `a` for autofigure what to do
- `target`(t) to search to bump for a specific node version (from #143)
- `F` to target a folder (was ran from `coorpacademy-lambda` root (from #143)
- `s` to `scope` in monorepo (scoped added in title and branch name) :new: a1faa744387b2f66b7465d4946cb6d62abb4c1e9
- `R` to set reviewer (from #143)

------

serverless+lerna support: Serverless runtime will be updated, and every package in a lerna mono repo will be updated :zap::dragon_face: 8fffb7f6365b9cc61e61caadffb3035a665eaad2 + 22dcf55f2c8b6487e4ec327c89579219da7f3b38

Some adjustement where also made for the program to be more flexible with the baseBranch, supporting non `master` branches

### CODE REVIEW

File: scripts/integration-test.sh [hunk 1]
```diff
@@ -39,4 +39,25 @@ if  [[ $current_commit != "$(git rev-parse HEAD)" ]]; then
 fi
 git --no-pager log --graph --decorate --pretty=oneline --abbrev-commit
 
-git checkout -b origins init
+# try a targeted bump with clean and extra command and token
```
Comment (StefanoA1) [inline scripts/integration-test.sh:44]:
```suggestion
PRE_COMMAND_FILE=".precommand"
POST_COMMAND_FILE=".postcommand"
# try a targeted bump with clean and extra command and token
```

File: scripts/integration-test.sh [hunk 2]
```diff
@@ -39,4 +39,25 @@ if  [[ $current_commit != "$(git rev-parse HEAD)" ]]; then
...
+echo "            " >> package.json
+npm run update -- upgrade --local --target 20 \
+  -fam "Beta test update-node major bump :scientist:" -R @Coorpacademy/development-mooc \
+  --clean -p "echo before > .precommand" -P "echo after > .postcommand"
```
Comment (StefanoA1) [inline scripts/integration-test.sh]:
```suggestion
  --clean -p "echo before > $PRE_COMMAND_FILE" -P "echo after > $POST_COMMAND_FILE"
```

File: scripts/integration-test.sh [hunk 3]
```diff
@@ -39,4 +39,25 @@ if  [[ $current_commit != "$(git rev-parse HEAD)" ]]; then
...
+# note: cannot test so far with a real branch. should set up a local upstream
+
+if ! git log --graph --decorate --pretty=oneline --abbrev-commit | grep -q "Upgrade Node to v20"; then
+    echo "Seems like no commit was created while it wasn't supposed to"
+    exit 2
+fi
+if  [[ ! -f .precommand ]]; then
```
Comment (StefanoA1) [inline scripts/integration-test.sh]:
```suggestion
if  [[ ! -f .$PRE_COMMAND_FILE ]]; then
```

File: scripts/integration-test.sh [hunk 4]
```diff
@@ -39,4 +39,25 @@ if  [[ $current_commit != "$(git rev-parse HEAD)" ]]; then
...
+    echo "Seems like pre command wasnt run"
+    exit 2
+fi
+if  [[ ! -f .postcommand ]]; then
```
Comment (StefanoA1) [inline scripts/integration-test.sh]:
```suggestion
if  [[ ! -f .$POST_COMMAND_FILE ]]; then
```

File: scripts/integration-test.sh [hunk 5]
```diff
@@ -39,4 +39,25 @@ if  [[ $current_commit != "$(git rev-parse HEAD)" ]]; then
...
+    echo "Seems like pre command wasnt run"
+    exit 2
+fi
+rm -f .precommand .postcommand
```
Comment (StefanoA1) [inline scripts/integration-test.sh]:
```suggestion
rm -f $PRE_COMMAND_FILE $POST_COMMAND_FILE
```

**Review by StefanoA1 — APPROVED** [review]

File: scripts/integration-test.sh (hunk 1 above)
Comment (AdrieanKhisbe) [reply scripts/integration-test.sh:44]: Bonne idée, je peux meme faire un mktemp pour etre sur fichier frai !

**Review by StefanoA1 — APPROVED** [review]

File: README.md
Comment (silouone) [inline README.md:1]: 👍 🙏

**Review by silouone — APPROVED** [review]

**Review by StefanoA1 — APPROVED** [review]

File: scripts/integration-test.sh [hunk 6]
```diff
@@ -39,4 +39,28 @@ if  [[ $current_commit != "$(git rev-parse HEAD)" ]]; then
...
+if ! git log --graph --decorate --pretty=oneline --abbrev-commit | grep -q "Upgrade Node to v20"; then
+    echo "Seems like no commit was created while it wasn't supposed to"
+    exit 2
+fi
+if  [[ ! -f $PRE_CLEAN_COMMAND_FILE ]]; then
+    echo "Seems like pre command wasnt run"
+    exit 2
+fi
+if  [[ ! -f $POST_CLEAN_COMMAND_FILE ]]; then
+    echo "Seems like pre command wasnt run"
```
Comment (StefanoA1) [inline scripts/integration-test.sh:61]:
```suggestion
    echo "Seems like post command wasnt run"
```

**Pull request merged**